python/sglang/srt/models/deepseek_v2.py

```diff
@@ -402,6 +402,41 @@ def handle_attention_aiter(attn, forward_batch):
 
 
 def handle_attention_nsa(attn, forward_batch):
+    """
+    Select MHA or MLA based on sequence length for optimal performance.
+
+    - Decode: MLA (avoids per-token decompression)
+    - Prefill <= 2048: MHA (topk ineffective, MHA has lower FLOPs)
+    - Prefill > 2048: MLA (topk filtering reduces computation significantly)
+
+    TODO: B200 (SM100) MHA path is temporarily disabled due to FA4 gpqa accuracy issues.
+    """
+    if forward_batch.forward_mode.is_decode_or_idle():
+        return AttnForwardMethod.MLA
+
+    if _is_extend_without_speculative(forward_batch):
+        NSA_THRESHOLD = envs.SGLANG_NSA_SEQ_LEN_THRESHOLD.value
```

Comment:
I don't think we should introduce an env variable for this purpose. It's set to 2048 because `self.nsa_index_topk` is 2048. We should just use `self.nsa_index_topk`.  If in another model, the `self.nsa_index_topk` value is set differently, this would have introduced a bug.